### PR TITLE
[Birthday] Remove hardcoded config calls and refactor

### LINKED ISSUE #414: [Birthday] Remove hardcoded config calls
#404.

### PR DESCRIPTION
**Pull request by Injabie3**
This PR will:
- Close #414 by using `get_attr` instead of hardcoding.
- Remove context manager use for single attribute changes: just use the set method directly.
- Update an incorrect comment.
- Replace `ctx.message.*` with the shortcut form `ctx.*`

There are no changes to existing behaviour.

### CODE REVIEW

File: cogs/birthday/birthday.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ async def _dailySweep(self):
             for guild in guilds:
                 # Make sure the guild is configured with birthday role.
                 # If it's not, skip over it.
-                bdayRoleId = await self.config.guild(guild).birthdayRole()
+                bdayRoleId = await self.config.guild(guild).get_atr(KEY_BDAY_ROLE)()
```
Comment (quachtridat) [inline cogs/birthday/birthday.py]: Typo here.
```suggestion
                bdayRoleId = await self.config.guild(guild).get_attr(KEY_BDAY_ROLE)()
```

**Comment by quachtridat** [comment]
###  Hardcoded config calls
Just one problem above. The rest of the changes LGTM.

### Possible refactors
Consider using
```py3
        await self.config.member(member).get_attr(KEY_IS_ASSIGNED).set(False)
```
for
[Injabie3:bday-config:birthday.py:330-331](https://github.com/SFUAnime/Ren/pull/432/files#diff-3286325c8014c66caaab082f23480f51dbd76ea6fcfa34076a99056d0c0a4d63R330)
```py3
        async with self.config.member(member).all() as userConfig:
            userConfig[KEY_IS_ASSIGNED] = False
```
and also
```py3
                                await self.config.member(member).get_attr(KEY_IS_ASSIGNED).set(True)
```
for
[Injabie3:bday-config:birthday.py:533-534](https://github.com/SFUAnime/Ren/pull/432/files#diff-3286325c8014c66caaab082f23480f51dbd76ea6fcfa34076a99056d0c0a4d63R533)
```py3
                                async with self.config.member(member).all() as memberConfig:
                                    memberConfig[KEY_IS_ASSIGNED] = True
```

<br>
<br>

[Injabie3:bday-config:birthday.py:156](https://github.com/SFUAnime/Ren/pull/432/files#diff-3286325c8014c66caaab082f23480f51dbd76ea6fcfa34076a99056d0c0a4d63R156)
```py3
        # If one of the parameters are missing, then raise error
```
Might be better if we write _send error messages_ rather than _raise error_ just so it is clear we are not making Ren throw exceptions here.

<br>
<br>

[Injabie3:bday-config:birthday.py:238](https://github.com/SFUAnime/Ren/pull/432/files#diff-3286325c8014c66caaab082f23480f51dbd76ea6fcfa34076a99056d0c0a4d63R238)
```py3
    async def list(self, ctx: Context):
```
I'd say we do something like `listBirthdays`  just to avoid `list` being a Python's built-in class.  Or, more like, it messes with my VS Code syntax highlighting ~~and I don't like that~~.

**Review by quachtridat — DISMISSED** [review]

**Comment by Injabie3** [comment]
Thanks, there's actually a few other things I want to refactor here anyways, but I wanted to keep them separate. I'll do it in the following commits, and do a rebase and merge.

**Review by quachtridat — APPROVED** [review]
So far LGTM.

Nit: add type hints to function args for consistency with how other function signatures look.


Line 63:
```py3
    async def setChannel(self, ctx, channel: discord.TextChannel = None):
```
to
```py3
    async def setChannel(self, ctx: Context, channel: discord.TextChannel = None):
```


Line 94:
```py3
    async def setRole(self, ctx, role: discord.Role):
```
to
```py3
    async def setRole(self, ctx: Context, role: discord.Role):
```


Line 120:
```py3
    async def test(self, ctx):
```
to
```py3
    async def test(self, ctx: Context):
```

**Pull request merged**